Sources/CoreUITestKit/XCUIElement+.swift

```diff
@@ -110,70 +111,105 @@ extension XCUIElement {
         let resourceName = self.description
         return XCTContext.runActivity(named: "Подождать, пока элемент \(String(describing: resourceName)) станет кликабельным") { _ in
             let expectation = XCTNSPredicateExpectation(
                 predicate: NSPredicate(format: "isHittable == true"),
                 object: self
             )
             let result = XCTWaiter.wait(for: [expectation], timeout: timeout)
-            let checkHittable = !self.dsl_isHittable || result == .completed
-            return checkHittable
+            return self.dsl_isHittable || (result == .completed)
         }
     }
-    
-    public func dsl_typeText(_ text: String, file: StaticString = #file, line: UInt = #line) {
+
+    public func dsl_typeText(_ text: String, needWait: Bool? = false, file: StaticString = #file, line: UInt = #line) {
         let resourceName = self.description
         XCTContext.runActivity(named: "Ввод текста \(text) в \(String(describing: resourceName))") { _ in
-            guard dsl_isHittable else {
-                XCTFail ("Элемент \(String(describing: resourceName)) не готов для нажатия",
-                file: file,
-                line: line
-                )
-                return
+            if needWait == true {
+                guard dsl_waitForHittable(timeout: 5) else {
+                    XCTFail("Элемент \(String(describing: resourceName)) не готов для нажатия",
+                            file: file,
+                            line: line
+                    )
+                    return
+                }
+                XCTContext.runActivity(named: "Ввод текста \(text)") { _ in
+                    typeText(text)
+                }
+            } else {
+                XCTContext.runActivity(named: "Ввод текста \(text)") { _ in
+                    typeText(text)
+                }
             }
-            typeText(text)
         }
     }
     
-    public func dsl_tap(file: StaticString = #file, line: UInt = #line) {
+    public func dsl_tap(needWait: Bool? = false, file: StaticString = #file, line: UInt = #line) {
         let resourceName = self.description
         XCTContext.runActivity(named: "Тап по элементу \(String(describing: resourceName))") { _ in
-            guard dsl_isHittable else {
-                XCTFail ("Элемент \(String(describing: resourceName)) не готов для нажатия",
-                file: file,
-                line: line
-                )
-                return
-            }
-            XCTContext.runActivity(named: "Тап") { _ in
-                tap()
+            if needWait == true {
+                guard dsl_waitForHittable(timeout: 5) else {
+                    XCTFail("Элемент \(String(describing: resourceName)) не готов для нажатия",
+                             file: file,
+                             line: line
+                    )
```

Comment:
```suggestion
                    XCTFail(
                        "Элемент \(String(describing: resourceName)) не готов для нажатия",
                        file: file,
                        line: line
                    )
```
обычно так переносим

Reply:
поправил, спасибо !